shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/test/java/org/apache/shardingsphere/sharding/distsql/update/DropShardingAlgorithmStatementUpdaterTest.java

```diff
@@ -53,16 +53,31 @@ public void assertCheckSQLStatementWithoutCurrentRule() throws RuleDefinitionVio
         updater.checkSQLStatement(shardingSphereMetaData, new DropShardingAlgorithmStatement(Collections.emptyList()), null);
     }
     
+    @Test(expected = RequiredAlgorithmMissedException.class)
+    public void assertCheckSQLStatementWithoutCurrentRuleWithIfExists() throws RuleDefinitionViolationException {
+        updater.checkSQLStatement(shardingSphereMetaData, new DropShardingAlgorithmStatement(true, Collections.emptyList()), null);
+    }
+    
```

Comment:
Should the expected result for this test be a pass?  Including the `IF EXISTS` keyword will only check if the algorithm is being used

Reply:
@lanchengx  will check `currentConfig` is null or else
![image](https://user-images.githubusercontent.com/35389145/155943575-217f9398-665c-4763-abdf-349b880e7013.png)

Reply:
Yes, I understand that the current configuration is checked for existence, but when the `IF EXIST` keyword is included (`containsExistClause` is true), there should be no need to check whether the current configuration exists. So the expected result of this test case should be passing instead of throwing an exception, right?
Are we considering adding a judgment for `containsExistClause` in the `checkCurrentRuleConfiguration()` method